### PR TITLE
refactor: Refactor all exceptions

### PR DESCRIPTION
**Pull request by sillyElvy**
## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [ ] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [ ] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

File: interactions/api/models/gw.py [hunk 1]
```diff
@@ -60,7 +61,9 @@ class ApplicationCommandPermissions(ClientSerializerMixin):
     application_id: Snowflake = field(converter=Snowflake)
     guild_id: Snowflake = field(converter=Snowflake)
     id: Snowflake = field(converter=Snowflake)
-    # permissions: List[Permission] = field(converter=convert_list(Permission))  # TODO fix circular import
+    # from ...client.models.command import Permission
+
+    # permissions: List[Permission] = field(converter=convert_list(Permission))
```
Comment (Toricane) [inline interactions/api/models/gw.py:66]: Is this supposed to be here?

File: interactions/api/models/misc.py [hunk 2]
```diff
@@ -222,8 +223,9 @@ def __init__(
     ):
 
         if not isinstance(filename, str):
-            raise TypeError(
-                "File's first parameter 'filename' must be a string, not " + str(type(filename))
+            raise LibraryException(
+                message=f"File's first parameter 'filename' must be a string, not + {str(type(filename))}",
```
Comment (Toricane) [inline interactions/api/models/misc.py]:
```suggestion
                message=f"File's first parameter 'filename' must be a string, not {str(type(filename))}",
```

**Review by Toricane — CHANGES_REQUESTED** [review]
A question and a suggestion

File: interactions/api/models/gw.py (hunk 1 above)
Comment (sillyElvy) [reply interactions/api/models/gw.py:66]: Yeah it was circular import and I attempted a fix lol

**Review by Toricane — APPROVED** [review]
LGTM!

**Pull request merged**